### PR TITLE
Add GPU to GENX conversion for several dialect operations 

### PR DESCRIPTION
**Pull request by etiotto**
GPU->GENX conversion for the following operations: 
  - gpu::ThreadIdOp -> `GENX::ThreadIdXOp, GENX::ThreadIdYOp, GENX::ThreadIdZOp
  - gpu::BlockIdOp -> GENX::BlockIdXOp, GENX::BlockIdYOp, GENX::BlockIdZOp
  - gpu::BlockDimOp -> GENX::BlockDimXOp, GENX::BlockDimYOp, GENX::BlockDimZOp
  - gpu::GridDimOp ->  GENX::GridDimXOp, GENX::GridDimYOp, GENX::GridDimZOp

### CODE REVIEW

**Comment by etiotto** [comment]
Depends on https://github.com/pengtu/llvm-project/pull/1

File: mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                                       GENX::ThreadIdYOp, GENX::ThreadIdZOp>>(
+          converter)
+      .add<GPUIndexIntrinsicOpLowering<gpu::BlockIdOp, GENX::BlockIdXOp,
+                                       GENX::BlockIdYOp, GENX::BlockIdZOp>>(
+          converter)
+      .add<GPUIndexIntrinsicOpLowering<gpu::BlockDimOp, GENX::BlockDimXOp,
+                                       GENX::BlockDimYOp, GENX::BlockDimZOp>,
+           GPUIndexIntrinsicOpLowering<gpu::GridDimOp, GENX::GridDimXOp,
+                                       GENX::GridDimYOp, GENX::GridDimZOp>>(
+          converter);
```
Comment (whitneywhtsang) [inline mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp]:
```suggestion
      .add<GPUIndexIntrinsicOpLowering<gpu::ThreadIdOp, GENX::ThreadIdXOp,
                                       GENX::ThreadIdYOp, GENX::ThreadIdZOp>,
           GPUIndexIntrinsicOpLowering<gpu::BlockIdOp, GENX::BlockIdXOp,
                                       GENX::BlockIdYOp, GENX::BlockIdZOp>,
           GPUIndexIntrinsicOpLowering<gpu::BlockDimOp, GENX::BlockDimXOp,
                                       GENX::BlockDimYOp, GENX::BlockDimZOp>,
           GPUIndexIntrinsicOpLowering<gpu::GridDimOp, GENX::GridDimXOp,
                                       GENX::GridDimYOp, GENX::GridDimZOp>>(
          converter);
```

**Review by whitneywhtsang — COMMENTED** [review]
only reviewed af3ada2443781593fd8b8de74b8090a7068dfab1.

File: mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp (hunk 1 above)
Comment (etiotto) [reply mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp]: done

**Review by whitneywhtsang — APPROVED** [review]

**Review by pengtu — APPROVED** [review]
Looks good, thank you!

**Pull request merged**